sklearn/feature_selection/sfs.py

```diff
@@ -0,0 +1,366 @@
+        Reducing this number can be useful to avoid an explosion of
+        memory consumption when more jobs get dispatched than CPUs can process.
+        This parameter can be:
+        - None, in which case all the jobs are immediately created and spawned.
+          Use this for lightweight and fast-running jobs,
+          to avoid delays due to on-demand spawning of the jobs
+        - An int, giving the exact number of total jobs that are spawned
+        - A string, giving an expression as a function
+            of `n_jobs`, as in `2*n_jobs`
+
+    Attributes
+    ----------
+    k_feature_idx_ : array-like, shape = [n_predictions]
+        Feature Indices of the selected feature subsets.
+
+    k_score_ : float
+        Cross validation average score of the selected subset.
+
+    subsets_ : dict
+        A dictionary of selected feature subsets during the
+        sequential selection, where the dictionary keys are
+        the lengths k of these feature subsets. The dictionary
+        values are dictionaries themselves with the following
+        keys: 'feature_idx' (tuple of indices of the feature subset)
+              'cv_scores' (list individual cross-validation scores)
+              'avg_score' (average cross-validation score)
+
+    Examples
+    --------
+    The following example shows how to use the sequential feature selector
+    with default settings to select a feature subset, consisting of
+    1 to 3 features, from iris. The selection criteria for this
+    feature subset is the average cross-validation performance
+    (cv=5 by default) of the `estimator` (here: KNN)
+    during the greedy forward selection search.
+
+        >>> from sklearn.feature_selection import SFS
+        >>> from sklearn.neighbors import KNeighborsClassifier
+        >>> from sklearn.datasets import load_iris
+        >>> iris = load_iris()
+        >>> X, y = iris.data, iris.target
+        >>> knn = KNeighborsClassifier(n_neighbors=3)
+        >>> sfs = SFS(knn, k_features=(1, 3))
+        >>> sfs = sfs.fit(X, y)
+        >>> round(sfs.k_score_, 4)
+        0.9733
+        >>> sfs.k_feature_idx_
+        (0, 2, 3)
+        >>> sfs.transform(X).shape
+        >>> (150, 3)
+
+    """
+    def __init__(self, estimator, k_features=1,
+                 forward=True, scoring=None,
+                 cv=5, n_jobs=1,
+                 pre_dispatch='2*n_jobs'):
+
+        self.estimator = clone(estimator)
+        self.k_features = k_features
+        self.forward = forward
+        self.pre_dispatch = pre_dispatch
+        self.scoring = scoring
+        if scoring is None:
+            if self.estimator._estimator_type == 'classifier':
+                scoring = 'accuracy'
+            elif self.estimator._estimator_type == 'regressor':
+                scoring = 'r2'
+            else:
+                raise ValueError('Estimator must '
+                                 'be a Classifier or Regressor.')
+
+        if isinstance(scoring, str):
```

Comment:
You could use the `check_scoring` from `sklearn.metrics.scorer ` in the `fit` method to get the `scorer`